spark/dl/src/main/scala/com/intel/analytics/bigdl/dataset/Transformer.scala

```diff
@@ -190,7 +209,8 @@ class SampleToBatch[T: ClassTag]
       private var labelData: Array[T] = null
       private val batchSize = batchSizePerPartition
 
-      private val sampleData = Array.tabulate(batchSize)(_ => Sample())
+      private val sampleData = Array.tabulate(batchSize)(_ =>
+        new TensorTensorSample(Tensor(), Tensor()))
```

Comment:
Why not Sample (Tensor(), Tensor())? We may even add a default (ore deprecated) Sample() method that calls Sample (Tensor(), Tensor()) to avoid user code changes